clients/go/pkg/commands/stream_jobs.go

```diff
@@ -0,0 +1,155 @@
+// limitations under the License.
+
+package commands
+
+import (
+	"context"
+	"io"
+	"log"
+	"time"
+
+	"github.com/camunda/zeebe/clients/go/v8/pkg/entities"
+	"github.com/camunda/zeebe/clients/go/v8/pkg/pb"
+)
+
+type StreamJobsConsumer chan<- entities.Job
+
+type DispatchStreamJobsCommand interface {
+	RequestTimeout(time.Duration) DispatchStreamJobsCommand
+	Send(ctx context.Context) error
+}
+
+type StreamJobsCommandStep1 interface {
+	JobType(string) StreamJobsCommandStep2
+}
+
+type StreamJobsCommandStep2 interface {
+	Consumer(StreamJobsConsumer) StreamJobsCommandStep3
+}
+
+type StreamJobsCommandStep3 interface {
+	DispatchStreamJobsCommand
+
+	TenantIds(...string) StreamJobsCommandStep3
+	Timeout(time.Duration) StreamJobsCommandStep3
+	WorkerName(string) StreamJobsCommandStep3
+	FetchVariables(...string) StreamJobsCommandStep3
+}
+
+type StreamJobsCommand struct {
+	Command
+	request        pb.StreamActivatedJobsRequest
+	consumer       StreamJobsConsumer
+	requestTimeout time.Duration
+}
+
+func (cmd *StreamJobsCommand) JobType(jobType string) StreamJobsCommandStep2 {
+	cmd.request.Type = jobType
+	return cmd
+}
+
+func (cmd *StreamJobsCommand) Consumer(consumer StreamJobsConsumer) StreamJobsCommandStep3 {
+	cmd.consumer = consumer
+	return cmd
+}
+
+func (cmd *StreamJobsCommand) TenantIds(tenantIds ...string) StreamJobsCommandStep3 {
+	cmd.request.TenantIds = tenantIds
+	return cmd
+}
+
+func (cmd *StreamJobsCommand) Timeout(timeout time.Duration) StreamJobsCommandStep3 {
+	cmd.request.Timeout = int64(timeout / time.Millisecond)
+	return cmd
+}
+
+func (cmd *StreamJobsCommand) WorkerName(workerName string) StreamJobsCommandStep3 {
+	cmd.request.Worker = workerName
+	return cmd
+}
+
+func (cmd *StreamJobsCommand) FetchVariables(fetchVariables ...string) StreamJobsCommandStep3 {
+	cmd.request.FetchVariable = fetchVariables
+	return cmd
+}
+
+func (cmd *StreamJobsCommand) RequestTimeout(requestTimeout time.Duration) DispatchStreamJobsCommand {
+	cmd.requestTimeout = requestTimeout
+	return cmd
+}
+
+func (cmd *StreamJobsCommand) Send(ctx context.Context) error {
+	if cmd.requestTimeout > 0 {
+		streamContext, cancelCtx := context.WithTimeout(ctx, cmd.requestTimeout)
+		ctx = streamContext
+		defer cancelCtx()
+	}
+
+	stream, err := cmd.openStream(ctx)
+	if err != nil {
+		return err
+	}
+
+	for {
+		job, err := stream.Recv()
+		if err == io.EOF {
+			break
+		}
+
+		if cmd.shouldRetry(ctx, err) {
+			stream, err = cmd.openStream(ctx)
+			if err != nil {
+				log.Printf("Failed to reopen job stream: %v\n", err)
+			}
+
+			continue
+		}
+
+		if err != nil {
+			return err
+		}
+
+		cmd.consumer <- entities.Job{ActivatedJob: job}
+	}
+
+	return nil
+}
+
+func (cmd *StreamJobsCommand) openStream(ctx context.Context) (pb.Gateway_StreamActivatedJobsClient, error) {
+	stream, err := cmd.gateway.StreamActivatedJobs(ctx, &cmd.request)
+
+	if err != nil {
+		if cmd.shouldRetry(ctx, err) {
+			return cmd.openStream(ctx)
```

Comment:
❓ can we loop endless here?

Reply:
Potentially yes, iff re-authentication succeeds but we keep getting `UNAUTHENTICATED` as a result. This affects most commands in the client though. AFAIK this also affects the Java client.